src/components/metadata/MetadataEditorModal.test.tsx

```diff
@@ -0,0 +1,107 @@
+import { MetadataEditorModal } from './MetadataEditorModal';
+import { schemaMock } from './MetadataEditorModal.stories';
+import { screen } from '@testing-library/dom';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+
+describe('MetadataEditorModal.tsx', () => {
+  test('component renders if open', () => {
+    render(
+      <MetadataEditorModal
+        handleCloseModal={jest.fn()}
+        isModalOpen={true}
+        name={'beans'}
+        schema={schemaMock.beans}
+      />,
+    );
+    const element = screen.queryByTestId('metadata-beans-modal');
+    expect(element).toBeInTheDocument();
+  });
+
+  test('component does not render if closed', () => {
+    render(
+      <MetadataEditorModal
+        handleCloseModal={jest.fn()}
+        isModalOpen={false}
+        name={'beans'}
+        schema={schemaMock.beans}
+      />,
+    );
+    const element = screen.queryByTestId('metadata-beans-modal');
+    expect(element).not.toBeInTheDocument();
+  });
+
+  test('editor works', async () => {
```

Comment:
I think that it makes sense to keep the test with mock schemas as you're doing right now, I was suggesting more like preparing the data that the component will need, then rendering it, and then ensuring that it looks like you would like to.

Pretty much as you have it right now, but instead of interacting with the UI to reach the desired point, already provide that data to the component and then make the assertions on top of it.

Reply:
So instead of testing everything, add/edit/delete in one place, test one action in one test, e.g. test add button and don't continue with the result of it, correct? I can do that.

Reply:
>  test one action in one test, e.g. test add button and don't continue with the result of it, correct? I can do that.

Exactly, just prepare the stage for the test, execute the action, and then do the evaluation. As AAA suggests.

Arrange, Act, and Assert.